src/sysctl/sysctl.c

```diff
@@ -51,19 +51,46 @@ static int apply_all(OrderedHashmap *sysctl_options) {
 
                 k = sysctl_write(property, value);
                 if (k < 0) {
-                        log_full_errno(k == -ENOENT ? LOG_INFO : LOG_WARNING, k,
-                                       "Couldn't write '%s' to '%s', ignoring: %m", value, property);
-
-                        if (r == 0 && k != -ENOENT)
-                                r = k;
+                        /* If the sysctl is not available in the kernel or we are running with reduced privileges and
+                         * cannot write it, then log about the issue at LOG_NOTICE level, and proceed without
+                         * failing. (EROFS is treated as a permission problem here, since that's how container managers
+                         * usually protected their sysctls.) In all other cases log an error and make the tool fail. */
+
+                        if (IN_SET(k, -EPERM, -EACCES, -EROFS, -ENOENT))
+                                log_notice_errno(k, "Couldn't write '%s' to '%s', ignoring: %m", value, property);
```

Comment:
@evverx well, I really wouldn't make a distinction here between EROFS/EPERM/EACCESS, which one you get with this is simply result of the security mechanism you pick. You can use r/o bind mounts, which means EROFS, but you could also implement something like this with selinux/aarmoer/smack/some future LSM or anything else. I think which one you use and hence which kind of permission check you see is an implementation detail and shouldn't alter our handling of it...